src/TimeSeriesFilter.cpp

```diff
@@ -400,7 +400,14 @@ time_t TimeSeriesFilter::timeBefore(time_t t) {
     return 0;
   }
   else if (this->clock()) {
-    return this->clock()->timeBefore(t);
+    auto b4 = this->clock()->timeBefore(t);
+    if (_resampleMode == ResampleModeLinear && source()->timeAfter(b4) == 0) {
```

Comment:
one weird edge case on this. 
consider a source series with times: `[100,200,300]`
... then take a resampler with clock (100s), and ask for `timeBefore(400)`
The resampler (`this`) will go to its clock, get time before 400, which is 300.
Then this new logic fires. Linear mode true, and `source()->timeAfter(300)` will be zero, since there are no values after t=300. But! we should be returning a point at 300, since resampler and source align perfectly at that time.

I think the existing convention for this sort of situation is to back up by 1s. 
so something like:

```
if (_resampleMode == ResampleModeLinear && source()->timeAfter(b4 - 1) == 0) {
```